### PR TITLE
Moving API to own app, adding interactive swagger docs

### PR DESCRIPTION
**Pull request by aphelionz**
You can see it in action here: https://park.hydroshare.org/hsapi/

### CODE REVIEW

**Comment by aphelionz** [comment]
@pkdash @alvacouch Please review at your earliest convenience. All tests are now passing.

**Comment by mjstealey** [comment]
@aphelionz - Looking forward to seeing this in swagger format. Once completed, we'll no longer need to maintain [hs_restclient](https://github.com/hydroshare/hs_restclient).

If the swagger.json/yaml file is a true reflection of the server code, then we can use swagger-codegen to create many clients in multiple languages auto-magically.

And since there is a docker implementation, we'll be able to do cool stuff like:
```bash
docker run --rm -v ${PWD}:/local swaggerapi/swagger-codegen-cli generate \
    -i https://URL/TO/swagger.json \
    -l python \
    -o /local/hs_restclient \
    -DpackageName=hs_restclient
```

And blamo, fully documented python hs_restclient

**Comment by aphelionz** [comment]
@mjstealey You can see it here https://park.hydroshare.org/hsapi/ or at least the beginnings of it

**Comment by aphelionz** [comment]
@mjstealey btw yes that is super cool. let's achieve that.

File: hs_rest_api/urls.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    # DEPRECATED: use from above instead
+    url(r'^scimeta/(?P<pk>[0-9a-f-]+)/$',
+        views.resource_rest_api.ScienceMetadataRetrieveUpdate.as_view(),
+        name='DEPRECATED_get_update_science_metadata'),
+
+    url(r'^resource/(?P<pk>[A-z0-9]+)/map/$',
+        views.resource_rest_api.ResourceMapRetrieve.as_view(),
+        name='get_resource_map'),
+
+    # Unused. See ResourceFileListCreate. This is now implemented there.
```
Comment (pkdash) [inline hs_rest_api/urls.py]: I think we can delete these commented code.

**Comment by pkdash** [comment]
@aphelionz Left one minor comment. Otherwise +1.

File: hs_rest_api/urls.py (hunk 1 above)
Comment (aphelionz) [inline hs_rest_api/urls.py]: Fixed!

**Pull request merged**